bundles/org.openhab.binding.lgthinq/src/main/java/org/openhab/binding/lgthinq/internal/handler/LGThinQAbstractDeviceHandler.java

```diff
@@ -0,0 +1,820 @@
+        final String channelUID;
+        final Command command;
+
+        public AsyncCommandParams(String channelUUID, Command command) {
+            this.channelUID = channelUUID;
+            this.command = command;
+        }
+    }
+
+    /**
+     * Returns the simple channel UID name, i.e., without group.
+     *
+     * @param uid Full UID name
+     * @return simple channel UID name, i.e., without group.
+     */
+    protected String getSimpleChannelUID(String uid) {
+        String simpleChannelUID;
+        if (uid.indexOf("#") > 0) {
+            // I have to remove the channelGroup from de channelUID
+            simpleChannelUID = uid.split("#")[1];
+        } else {
+            simpleChannelUID = uid;
+        }
+        return simpleChannelUID;
+    }
+
+    /**
+     * Return empty string if null argument is passed
+     *
+     * @param value value to test
+     * @return empty string if null argument is passed
+     */
+    protected final String emptyIfNull(@Nullable String value) {
+        return Objects.requireNonNullElse(value, "");
+    }
+
+    /**
+     * Return the key informed if there is no correpondent value in map for that key.
+     *
+     * @param map map with key/value
+     * @param key key to search for a value into map
+     * @return return value related to that key in the map, or the own key if there is no correspondent.
+     */
+    protected final String keyIfValueNotFound(Map<String, String> map, String key) {
+        return Objects.requireNonNullElse(map.get(key), key);
+    }
+
+    @SuppressWarnings("null")
+    protected void startCommandExecutorQueueJob() {
+        if (commandExecutorQueueJob == null || commandExecutorQueueJob.isDone()) {
+            commandExecutorQueueJob = getExecutorService().submit(getQueuedCommandExecutor());
+        }
+    }
+
+    @SuppressWarnings("null")
+    protected void stopCommandExecutorQueueJob() {
+        if (commandExecutorQueueJob != null) {
+            commandExecutorQueueJob.cancel(true);
+        }
+        commandExecutorQueueJob = null;
+    }
+
+    @SuppressWarnings("null")
+    protected void handleStatusChanged(ThingStatus newStatus, ThingStatusDetail statusDetail) {
+        if (lastThingStatus != ThingStatus.ONLINE && newStatus == ThingStatus.ONLINE) {
+            // start the thing polling
+            startThingStatePolling();
+        } else if (lastThingStatus == ThingStatus.ONLINE && newStatus == ThingStatus.OFFLINE
+                && BRIDGE_STATUS_DETAIL_ERROR.contains(statusDetail)) {
+            // comunication error is not a specific Bridge error, then we must analise it to give
+            // this thinq the change to recovery from communication errors
+            if (statusDetail != ThingStatusDetail.COMMUNICATION_ERROR
+                    || (getBridge() != null && getBridge().getStatus() != ThingStatus.ONLINE)) {
+                // in case of status offline, I only stop the polling if is not an COMMUNICATION_ERROR or if
```

Comment:
This would make it possible to remove the null suppresion.
```suggestion
                    || (getBridge() instanceof Bridge bridge && bridge.getStatus() != ThingStatus.ONLINE)) {
```

Reply:
Good One ! Tnx.